src/cheri_insts.sail

```diff
@@ -403,7 +400,7 @@ union clause ast = CSetFlags : (regidx, regidx, regidx)
  */
 function clause execute(CSetFlags(cd, cs1, rs2)) = {
   let cs1_val' = C(cs1);
-  let cs1_val = {cs1_val' with tag = cs1_val'.tag & not(isCapSealed(cs1_val'))};
+  let cs1_val = invalidateCapUnless(cs1_val', not(isCapSealed(cs1_val')));
```

Comment:
I personally find this easier to read as `invalidateCapIf(cs1_val', isCapSealed(cs1_val'))`. No strong preference either way though.

Reply:
Agreed, the double negative is a bit confusing.

Reply:
Yep, I was thinking of it as "the properties that must be true after this for the tag to be set", but it just confused things. I've changed to `clearTagIf`, and negated (or unnegated) the conditions.